pkg/operation/botanist/component/kubeapiserver/deployment.go

```diff
@@ -724,181 +722,17 @@ func (k *kubeAPIServer) handleVPNSettings(
-			MountPath: volumeMountPathLibModules,
-		}},
-	}}
-	deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
-		Name: volumeNameLibModules,
-		VolumeSource: corev1.VolumeSource{
-			HostPath: &corev1.HostPathVolumeSource{Path: "/lib/modules"},
-		},
-	})
-
-	vpnSeedContainer := corev1.Container{
-		Name:            containerNameVPNSeed,
-		Image:           k.values.Images.VPNSeed,
-		ImagePullPolicy: corev1.PullIfNotPresent,
-		Env: []corev1.EnvVar{
-			{
-				Name:  "MAIN_VPN_SEED",
-				Value: "true",
-			},
-			{
-				Name:  "OPENVPN_PORT",
-				Value: "4314",
-			},
-			{
-				Name:  "APISERVER_AUTH_MODE",
-				Value: "client-cert",
-			},
-			{
-				Name:  "APISERVER_AUTH_MODE_CLIENT_CERT_CA",
-				Value: volumeMountPathCA + "/" + secrets.DataKeyCertificateBundle,
-			},
-			{
-				Name:  "APISERVER_AUTH_MODE_CLIENT_CERT_CRT",
-				Value: volumeMountPathVPNSeed + "/" + secrets.DataKeyCertificate,
-			},
-			{
-				Name:  "APISERVER_AUTH_MODE_CLIENT_CERT_KEY",
-				Value: volumeMountPathVPNSeed + "/" + secrets.DataKeyPrivateKey,
-			},
-			{
-				Name:  "SERVICE_NETWORK",
-				Value: k.values.VPN.ServiceNetworkCIDR,
-			},
-			{
-				Name:  "POD_NETWORK",
-				Value: k.values.VPN.PodNetworkCIDR,
-			},
-		},
-		Resources: corev1.ResourceRequirements{
-			Requests: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("100m"),
-				corev1.ResourceMemory: resource.MustParse("128Mi"),
-			},
-			Limits: corev1.ResourceList{
-				corev1.ResourceMemory: resource.MustParse("128Mi"),
-			},
-		},
-		SecurityContext: &corev1.SecurityContext{
-			Capabilities: &corev1.Capabilities{
-				Add: []corev1.Capability{"NET_ADMIN"},
-			},
-			Privileged: pointer.Bool(true),
-		},
-		TerminationMessagePath:   corev1.TerminationMessagePathDefault,
-		TerminationMessagePolicy: corev1.TerminationMessageReadFile,
-		VolumeMounts: []corev1.VolumeMount{
-			{
-				Name:      volumeNameVPNSeed,
-				MountPath: volumeMountPathVPNSeed,
-			},
-			{
-				Name:      volumeNameVPNSeedTLSAuth,
-				MountPath: volumeMountPathVPNSeedTLSAuth,
-			},
-			{
-				Name:      volumeNameCA,
-				MountPath: volumeMountPathCA,
-			},
-		},
-	}
-
-	if k.values.VPN.NodeNetworkCIDR != nil {
-		vpnSeedContainer.Env = append(vpnSeedContainer.Env, corev1.EnvVar{
-			Name:  "NODE_NETWORK",
-			Value: *k.values.VPN.NodeNetworkCIDR,
-		})
-	}
-
-	deployment.Spec.Template.Spec.Containers = append(deployment.Spec.Template.Spec.Containers, vpnSeedContainer)
-	deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, []corev1.Volume{
-		{
-			Name: volumeNameVPNSeed,
-			VolumeSource: corev1.VolumeSource{
-				Projected: &corev1.ProjectedVolumeSource{
-					DefaultMode: pointer.Int32(400),
-					Sources: []corev1.VolumeProjection{
-						{
-							Secret: &corev1.SecretProjection{
-								LocalObjectReference: corev1.LocalObjectReference{
-									Name: secretLegacyVPNCAClient.Name,
```

Comment:
Can we now drop https://github.com/gardener/gardener/blob/44246e18b204925bc8fe782ef978ab1cdfa4066c/pkg/operation/botanist/component/kubeapiserver/deployment.go#L724?